### ISSUE TITLE
Failing to deploy own build of LibreChat

### ISSUE THREAD

**Issue by larsrinn** [thunderbug1/LibreChatAzureDeployment#3]
First of all, thanks for preparing all of this. This has helped me to deploy LibreChat much quicker. However, I'm struggling to change the image to one from an internal registry. Do you have an idea why this is the case? I managed to set everything up:

- Push to our container registry
- The web app deployment pulls from that registry

Now, during startup, we get the error logs:

```
2023-08-21T11:06:10.961Z INFO  - docker run -d --expose=80 --name librechatapp859rfpgz_1_ce5fcae7 -e PORT=80 -e WEBSITES_ENABLE_APP_SERVICE_STORAGE=false -e WEBSITES_PORT=80 -e WEBSITE_SITE_NAME=librechatapp859rfpgz -e WEBSITE_AUTH_ENABLED=False -e WEBSITE_ROLE_INSTANCE_ID=0 -e WEBSITE_HOSTNAME=librechatapp859rfpgz.azurewebsites.net -e WEBSITE_INSTANCE_ID=1944b9e704a67460698ada12dadb81e86917d687076fd3f004bf1d7a730f9805 -e HTTP_LOGGING_ENABLED=1 -e WEBSITE_USE_DIAGNOSTIC_SERVER=False nodeenergy.azurecr.io/librechat:node.energy  

2023-08-21T11:06:13.962Z INFO  - Initiating warmup request to container librechatapp859rfpgz_1_ce5fcae7_msiProxy for site librechatapp859rfpgz
2023-08-21T11:06:14.055Z INFO  - Container librechatapp859rfpgz_1_ce5fcae7_msiProxy for site librechatapp859rfpgz initialized successfully and is ready to serve requests.
2023-08-21T11:06:14.056Z INFO  - Initiating warmup request to container librechatapp859rfpgz_1_ce5fcae7 for site librechatapp859rfpgz
2023-08-21T11:06:14.068Z ERROR - Container librechatapp859rfpgz_1_ce5fcae7 for site librechatapp859rfpgz has exited, failing site start
2023-08-21T11:06:14.077Z ERROR - Container librechatapp859rfpgz_1_ce5fcae7 didn't respond to HTTP pings on port: 80, failing site start. See container logs for debugging.
```

I have already tried to change the setting `WEBSITES_PORT` to 3080 but no success so far.

---

The command I'm using to build the image is:

```
docker build --tag $IMAGE_NAME --file Dockerfile.multi --target api-build .
```

(inspired from line 37 here: https://github.com/danny-avila/LibreChat/blob/main/.github/workflows/dev-images.yml )

**Comment by thunderbug1** [comment]
did you change PORT   +   WEBSITES_PORT    +   configured the webapp to listen to the configured port?
Everything needs to fit together for a working system.

**Comment by larsrinn** [comment]
My assumption was: If I take the `docker build` command from the GitHub action, this is the which builds the images which are used. So the images should be identical (apart from the slight modification in the code I made, of course). So I should be able to just swap the images without changing anything else. 

Changing the `WEBSITES_PORT` to 3080 was just an idea to check whether this error message will disappear. I came up with it, because this is the exposed port.

Do you have an idea why port 80 is working for the `ghcr.io/danny-avila/librechat-dev-api:latest`?

**Comment by thunderbug1** [comment]
port 80 is the default port for azure web apps, so unless you explicitly change it, it defaults to that.

**Comment by larsrinn** [comment]
I think I have a decent knowledge in Docker and DevOps but am really struggling here. Can you explain how this makes sense:

```
DOCKER_REGISTRY_SERVER_URL          = "https://index.docker.io"
DOCKER_CUSTOM_IMAGE_NAME            = "ghcr.io/danny-avila/librechat-dev-api:latest"
```

The image name let's me presume the image is hosted on GitHub's container registry. But it claims it should use the one from Dockerhub. On Dockerhub, I don't find an image with a matching name.

Also, did you see the note on moving these settings to the application stack block in the Azure provider documentation?

<img width="613" alt="grafik" src="https://github.com/thunderbug1/LibreChatAzureDeployment/assets/22575361/edc58271-25d3-4c4d-89cc-1a9bf13a415d">

**Comment by thunderbug1** [comment]
Yeah It might very well be that DOCKER_REGISTRY_SERVER_URL is actually superfluous or on the wrong place here. I didn't really look into that since I didn't have a problem with that.

**Comment by Pwd9000-ML** [comment]
I have the same issue, and for the life of me cannot figure it out!
All the ports are set to 80 but the container just wont start up. Got the same diagnostic error cannot respond to ping son HTTP port 80

**Comment by thunderbug1** [comment]
debugging failing docker image starts can be quite tricky in azure webapps since you don't get any useful logs out of it.
Can you try to run the docker image locally with the same environment variables to see if it starts there?
maybe the latest version had a breaking change at some point and doesn't start because of that.

**Comment by Pwd9000-ML** [comment]
I get these errors:
App error: 

`2024-01-22T11:51:05.151465431Z 2024-01-22 11:51:05 [31merror[39m: [31mThere was an uncaught error: The Authorization header is missing. It must use the bearer authorization method.[39m`

and platform error:

`2024-01-22T11:50:45.350Z INFO  - docker run -d --expose=80 --name librechatapp2698_1_00260834 -e WEBSITE_USE_DIAGNOSTIC_SERVER=false -e DOCKER_CUSTOM_IMAGE_NAME=ghcr.io/danny-avila/librechat-dev-api:latest -e PORT=80 -e WEBSITES_ENABLE_APP_SERVICE_STORAGE=false -e WEBSITES_PORT=80 -e WEBSITE_SITE_NAME=librechatapp2698 -e WEBSITE_AUTH_ENABLED=False -e WEBSITE_ROLE_INSTANCE_ID=0 -e WEBSITE_HOSTNAME=librechatapp2698.azurewebsites.net -e WEBSITE_INSTANCE_ID=ca73f7ceb97227cd443fc5fb5ee6f8bbafbaa7f2ca5a209c491ad2b7e83c8c35 -e HTTP_LOGGING_ENABLED=1 ghcr.io/danny-avila/librechat-dev-api:latest 
2024-01-22T11:50:49.126Z INFO  - Initiating warmup request to container librechatapp2698_1_00260834_msiProxy for site librechatapp2698
2024-01-22T11:50:49.282Z INFO  - Container librechatapp2698_1_00260834_msiProxy for site librechatapp2698 initialized successfully and is ready to serve requests.
2024-01-22T11:50:49.290Z INFO  - Initiating warmup request to container librechatapp2698_1_00260834 for site librechatapp2698
2024-01-22T11:51:05.650Z INFO  - Waiting for response to warmup request for container librechatapp2698_1_00260834. Elapsed time = 16.5240623 sec
2024-01-22T11:51:05.842Z ERROR - Container librechatapp2698_1_00260834 for site librechatapp2698 has exited, failing site start
2024-01-22T11:51:05.864Z ERROR - Container librechatapp2698_1_00260834 didn't respond to HTTP pings on port: 80, failing site start. See container logs for debugging.
2024-01-22T11:51:06.152Z INFO  - Stopping site librechatapp2698 because it failed during startup.`

**Comment by Pwd9000-ML** [comment]
UPDATE: When I remove Meilisearch integration the app works fine, I feel that the integration of Meilisearch in the Libre app integration may be what's causing the issues
